src/OrleansTelemetryConsumers.DataDog/DataDogTelemetryConsumer.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Newtonsoft.Json;
+using Orleans.Runtime;
+using StatsdClient;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Orleans.TelemetryConsumers.DataDog
+{
+    public class DataDogTelemetryConsumer : IMetricTelemetryConsumer, IEventTelemetryConsumer
+    {
+        public DataDogTelemetryConsumer(string statsdServerName = "127.0.0.1", int statsdPort = 8125, string prefix = "")
+        {
+            var dogstatsdConfig = new StatsdConfig
+            {
+                StatsdServerName = statsdServerName,
+                StatsdPort = statsdPort,
+                Prefix = prefix
+            };
+
+            DogStatsd.Configure(dogstatsdConfig);
+        }
+
+        public void DecrementMetric(string name)
+        {
+            DogStatsd.Decrement(name);
+        }
+
+        public void DecrementMetric(string name, double value)
+        {
+            DogStatsd.Decrement(name, (int)value);
+        }
+
+        public void IncrementMetric(string name)
+        {
+            DogStatsd.Increment(name);
+        }
+
+        public void IncrementMetric(string name, double value)
+        {
+            DogStatsd.Increment(name, (int)value);
+        }
+
+        public void TrackMetric(string name, TimeSpan value, IDictionary<string, string> properties = null)
+        {
+            DogStatsd.Increment(name, (int)value.TotalMilliseconds, tags: ToStringArray(properties));
```

Comment:
I meant just a comment in the code to explain the cast is always expected to be in valid range and chopping off fractionals is OK etc. To assure readers after us, or possibly alert them our reasoning was not sound. :)